### PR TITLE
feat(bazel): announce version/command per spawn; unify setup phase + collapse lifecycle to one hook

### PR DESCRIPTION
**Pull request by gregmagolan**
Improves pre-task / bazel-spawn diagnostics and unifies the pre-task lifecycle.

## 1. Announce the Bazel version and command on each spawn

Before each `bazel build`/`test`/`run` spawn, optionally print the detected Bazel version and the exact command line. The existing `--announce-rc` is renamed to `--announce-bazel-rc` so the three form one family, all `"auto" | "true" | "false"` defaulting to `auto` (shown on CI, quiet locally):

| Flag | Prints | Default |
|---|---|---|
| `--announce-bazel-rc` | resolved `.bazelrc` flags (was `--announce-rc`) | `auto` |
| `--announce-bazel-version` | `INFO: Bazel <version>` | `auto` |
| `--announce-bazel-command` | `INFO: Spawning: <command>` | `auto` |

- `auto` → on when a CI host is detected, off locally; `true`/`false` force it.
- The command and the rc disclosure are scrubbed through the existing redaction (`stream::redaction`): env-var values, request headers (header name kept, value hidden), and URL credentials become `<REDACTED>`.
- `--announce-bazel-rc` displays each rc source path relative to the most meaningful anchor: `./` under the workspace root, `~/` under `$HOME`, else absolute.
- A non-release Bazel prints `Bazel development version (version-conditional flags assume latest)`.
- The three flags are declared once by a shared `announce_bazel_args` helper that each Bazel-spawning task splats into its args.

## 2. Unify the pre-task lifecycle into one `setup` phase

`setup_phase` and `preflight_phase` are collapsed into a single `setup_phase` that every task opens with, locally and on CI, passing its result `kind` + `data`: append runner-job-history, open the `🔧 Setup` phase mark (the task's first `task_update` — it inits the status surfaces **and** renders their first body from `kind`/`data`), parse the `.bazelrc` + resolve flags (when given a `BazelTrait`), then run the `health_check` hooks. Bazel work is gated on the `BazelTrait` argument so a non-Bazel task can use the same phase.

`HealthCheckTrait` is also collapsed to a single `health_check` hook (was `pre_health_check` + `post_health_check`). A hook reporting an unhealthy environment now **fails the task** from inside `setup_phase`: it concludes the status surface (terminal "Health check failed") then aborts with the hook's detailed reason. Previously an unhealthy Bazel server signaled the runner but let the task proceed — a latent bug this fixes.

## 3. Collapse the lifecycle to a single `task_update` hook

`TaskLifecycleTrait` had three hooks — `task_started`, `task_update`, `task_complete`. `task_complete` had no subscribers (dead), and `task_started` only ever did one-time surface init. Both are removed, leaving **one** hook:

- A handler's **first** `task_update` is its cue to initialize (authenticate, create the GitHub check run / first Buildkite annotation). The first update is the `🔧 Setup` phase mark from `setup_phase`, which also renders the first surface body.
- The **last** update carries `final = True`; handlers conclude their surface there.
- The task subject (target patterns / formatter target / …) rides on the `TaskUpdate` (new `subject` field, named `subject` end-to-end). `setup_phase` sets it on the first update; a later update may refine it (e.g. `delivery` emits its resolved targets once a query resolves). Handlers latch the last non-empty value, so the surface title updates in place. An empty subject renders empty (no `//...` placeholder); the per-task `summary_title` helpers share one `truncate_for_title` so a long target list is capped identically.

This removes the ordering constraint between the (now gone) `task_started` and the first `task_update`. It also drops the startup `GET /rate_limit` probe: the first status-check API response already seeds the same App-bucket quota state, the first call is never throttle-gated, and the cold-start floor makes a zero- vs one-observation buffer identical — so the probe only cost a startup round-trip.

## 4. Timing-breakdown alignment

Pad the phase label column by display width (emoji = 2 cells) instead of codepoint count, and use a single-astral setup emoji (`🔧`) that renders two cells like the other phase emoji.

---

### Changes are visible to end-users: yes

- Searched for relevant documentation and updated as needed: yes
- Breaking change (forces users to change their own code or config): yes
- Suggested release notes appear below: yes

### Suggested release notes

- Renamed `--announce-rc` to `--announce-bazel-rc`; added `--announce-bazel-version` and `--announce-bazel-command`. All three default to `auto` (shown on CI, quiet locally). The command and rc disclosures redact secrets and shorten rc paths.
- The separate "Pre-flight" task phase is merged into "Setup"; all pre-task work now reports under one phase, shown locally and on CI.
- A failed Bazel server health check now fails the task (and reports "Health check failed" on the status surface) instead of letting it proceed.

### Test plan

- Rust: `engine::bazel` build (19) + redaction (26), `bazelrc` (83, incl. path-anchor edge cases), `eval::multi_phase` (2), `task_info`.
- AXL: full `aspect tests axl`; template-snapshot dev tests (bazel-results, lint, format, gazelle, delivery — incl. a long-subject title-capping assertion, bk-annotation — incl. an empty-subject scenario); `test-bazel-flags` (incl. `announce_bazel_args`), `test-health-check-ready` (incl. `run_health_checks` first-error-wins), `test-rate-limit`, `test-github-detect`.
- Manual: local `aspect <task>` drives `setup_phase` → first `task_update` → terminal; `🔧 Setup` row aligns with `🔨 Build`; `--announce-bazel-version=true` prints the version line; an injected unhealthy `health_check` hook makes setup_phase render a "Health check failed" terminal and fail the task with the detailed reason.
- CI

### CODE REVIEW

File: crates/aspect-cli/src/builtins/aspect/lib/bazel_flags.axl [hunk 1]
```diff
@@ -133,9 +133,25 @@ def setup_bazel_command(ctx, command, bazel_trait, base_flags = []):
     rc_startup_flags, expanded_flags = rc.expand_all(command = command)
     ctx.bazel.startup_flags.extend(startup_flags + rc_startup_flags + ["--ignore_all_rc_files"])
 
+    _announce_bazel_version(ctx)
```
[the remark this answers is not in the document]
Comment (gregmagolan) [reply crates/aspect-cli/src/builtins/aspect/lib/bazel_flags.axl]: Resolved
